tests/test_rpm_verifier.py

```diff
@@ -15,178 +15,164 @@
     ProcessedImage,
     generate_output,
     get_rpmdb,
+    get_rpms_data,
     get_unsigned_rpms,
-    parse_image_input,
 )
 
 
+def test_get_rpmdb(tmp_path: Path) -> None:
+    """Test get_rpmdb"""
+    image = "my-image"
+    mock_runner = MagicMock()
+    out = get_rpmdb(
+        container_image=image,
+        target_dir=tmp_path,
+        runner=mock_runner,
+    )
+    assert mock_runner.call_count == 1
+    assert out == tmp_path
+
+
 @pytest.mark.parametrize(
-    ("processed_images", "expected_print", "expected_failures_exist"),
+    ("test_input", "expected"),
     [
         pytest.param(
-            [ProcessedImage(image="i1", unsigned_rpms=[])],
-            "No unsigned RPMs found.",
-            False,
-            id="No unsigned RPMs + no errors. Do not report failures",
-        ),
-        pytest.param(
-            [ProcessedImage(image="i1", unsigned_rpms=["my-rpm"])],
             dedent(
                 """
-                Found unsigned RPMs:
-                i1: my-rpm
+                libssh-config-0.9.6-10.el8_8 RSA/SHA256, Tue 6 May , Key ID 1234567890
+                python39-twisted-23.10.0-1.el8ap (none)
+                libmodulemd-2.13.0-1.el8 RSA/SHA256, Wed 18 Aug , Key ID 1234567890
+                gpg-pubkey-d4082792-5b32db75 (none)
                 """
             ).strip(),
-            True,
-            id="Unsigned RPM + no errors. Report failures",
+            [
+                "libssh-config-0.9.6-10.el8_8 RSA/SHA256, Tue 6 May , Key ID 1234567890",
+                "python39-twisted-23.10.0-1.el8ap (none)",
+                "libmodulemd-2.13.0-1.el8 RSA/SHA256, Wed 18 Aug , Key ID 1234567890",
+                "gpg-pubkey-d4082792-5b32db75 (none)",
+            ],
+            id="Mix of signed and unsigned",
         ),
+        pytest.param("", [], id="Empty list"),
+    ],
+)
+def test_get_rpms_data(test_input: list[str], expected: list[str]) -> None:
```

Comment:
Currently, we're just testing that we can split a string, which is not much.
I think it can be beneficial if we assert that we're making the expected call to the runner.

Maybe add another assertion such as:
```python
assert mock_runner.call_args == expected_runner_call
assert mock_runner.call_count == 1
```

You can also use [assert_called_once_with](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_called_once_with) that should look nicer, but it usually should be used when creating the mock with `create_autospec`.

The reason being that when creating a general mock, it will allow any method to be called on it and will return something that translates to boolean `True`, and in some circumstances, you might think you're calling that method and getting a True assertion, but in fact you're calling some other non-existing function and get positive result, because it will always return a positive result. `create_autospec` limits the mock to only have the methods that it should.

I think you could have something like this:

 ```python
    mock_runner = create_autospec(run, return_value=test_input)
    expected_runner_call = ...
    result = get_rpms_data(rpmdb=Path("rpmdb_folder"), runner=mock_runner)
    mock_runner.assert_called_once_with(expected_runner_call)
```

Reply:
I added it, and also had to add
`# pylint: disable=duplicate-code` in `rpm_verifier` because I got errors from pylint

Reply:
I imagine it's happening as it's both in code and in test. Can you move that comment to the test rather than the code?

Reply:
I tried to, the only place it accepts the comment is where I put it